slang-api/src/test/java/com/sonarsource/slang/impl/ParameterTreeImplTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.impl;
+
+import com.sonarsource.slang.api.IdentifierTree;
+import com.sonarsource.slang.api.NativeKind;
+import com.sonarsource.slang.api.Tree;
+import com.sonarsource.slang.api.TreeMetaData;
+import org.junit.Test;
+
+import static com.sonarsource.slang.utils.SyntacticEquivalence.areEquivalent;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ParameterTreeImplTest {
+
+  private class TypeNativeKind implements NativeKind {}
+
+  @Test
+  public void test() {
+    TreeMetaData meta = null;
+    Tree parameterType = new NativeTreeImpl(meta, new TypeNativeKind(), null);
+    IdentifierTree identifierTreeX = new IdentifierTreeImpl(meta, "x");
+    IdentifierTree identifierTreeY = new IdentifierTreeImpl(meta, "y");
+    ParameterTreeImpl parameterTreeX = new ParameterTreeImpl(meta, identifierTreeX, null);
+    ParameterTreeImpl parameterTreeXTyped = new ParameterTreeImpl(meta, identifierTreeX, parameterType);
+    ParameterTreeImpl parameterTreeY = new ParameterTreeImpl(meta, identifierTreeY, parameterType);
+
+    assertThat(parameterTreeX.children()).hasSize(2);
+    assertThat(parameterTreeX.type()).isNull();
+    assertThat(parameterTreeX.identifier()).isEqualTo(identifierTreeX);
+    assertThat(areEquivalent(parameterTreeX, parameterTreeX)).isTrue();
```

Comment:
I'd prefer to have another instance here `new IdentifierTreeImpl(meta, "x")` for the second tree, to ensure equivalence works on values of this new tree type and not only on references.